server/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -184,9 +190,14 @@ private Setting(
             if (propertiesAsSet.contains(Property.Deprecated) && propertiesAsSet.contains(Property.DeprecatedWarning)) {
                 throw new IllegalArgumentException("setting [" + key + "] cannot be deprecated at both critical and warning levels");
             }
+            if ((propertiesAsSet.contains(Property.Deprecated) || propertiesAsSet.contains(Property.DeprecatedWarning))
+                && propertiesAsSet.contains(Property.IndexSettingDeprecatedInV7AndRemovedInV8)) {
+                throw new IllegalArgumentException("setting [" + key + "] cannot be just deprecated and deprecated and removed in V7");
+            }
```

Comment:
Hmm suggestion got lost it seems. Here it is again:

```suggestion
            if (propertiesAsSet.stream().filter(DEPRECATED_PROPERTIES::contains).count() > 1) {
                throw new IllegalArgumentException("setting [" + key + "] must be at most one of [" + DEPRECATED_PROPERTIES + "]");
            }
```

Reply:
That's a really nice code simplification. I'll add it in.